packages/client/src/objectSet/createObjectSet.ts

```diff
@@ -41,7 +38,23 @@ function isObjectTypeDefinition(
   return def.type === "object";
 }
 
-const searchAroundPrefix = "searchAround_";
+export function isObjectSet(o: any): o is ObjectSet<any> {
+  return o != null && typeof o === "object"
+    && isWireObjectSet(objectSetDefinitions.get(o));
```

Comment:
We can probably just shortcut this whole thing to `objectSetDefinitions.get(o) != null`